src/client/Client.cc

```diff
@@ -10586,6 +10586,7 @@ Inode *Client::open_snapdir(Inode *diri)
     in->mode = diri->mode;
     in->uid = diri->uid;
     in->gid = diri->gid;
+    in->nlink = 1;
```

Comment:
I assume this is set to 0 otherwise? Ok, if so, but we should note that ganesha currently does not handle snapshots well. It's quite likely to produce ESTALE errors if the ganesha server is restarted with active clients. From FSAL_CEPH create_handle() routine:
```
        /* Check our local cache first */
        i = ceph_ll_get_inode(export->cmount, key->chk_vi);
        if (!i) {
                /*
                 * Try the slow way, may not be in cache now.
                 *
                 * Currently, there is no interface for looking up a snapped
                 * inode, so we just bail here in that case.
                 */
                if (key->chk_vi.snapid.val != CEPH_NOSNAP)
                        return ceph2fsal_error(-ESTALE);

                rc = ceph_ll_lookup_inode(export->cmount, key->chk_vi.ino, &i);
                if (rc)
                        return ceph2fsal_error(rc);
        }
```
The problem is that ceph_ll_lookup_inode doesn't deal with snapids. We'd need an equivalent to that in libcephfs + wire up ganesha to use it.